docs/dApps.mdx

```diff
@@ -0,0 +1,52 @@
+---
+title: Decentralized applications (dApps)
+authors: Benjamin Pilia, Tim McMackin
+last_update:
+  date: 12 January 2024
+---
+
+import LucidDiagram from '@site/src/components/LucidDiagram';
+
+One of the main features of blockchains is _decentralization_: each transaction is verified by multiple nodes and its validation process does not rely on a single trusted third party.
+Decentralized applications (dApps or Dapps) take advantage of these features to create applications that are independent, transparent, and trustless.
+
+In general, dApps have these parts:
+
+- **Frontend**: An off-chain component that can act as a user interface to simplify interaction with the on-chain component, run off-chain processing, and get information from sources that are not available to the on-chain component
+- **Middleware**: Optionally, an [indexer](./developing/information/indexers) to interpret the backend information and provide it in a more convenient format for the front-end component
+- **Backend**: An on-chain component that consists of one or more [smart contracts](./smart-contracts)
+
+The off-chain component can be nearly any kind of program, including a web application, mobile or desktop app, or command-line interface.
+It relies on wallets and tools to interact with the smart contract on behalf of a user's Tezos account.
+
+<LucidDiagram width="640px" height="480px" src="https://lucid.app/documents/embedded/8caf9ef1-11e4-454a-bbb6-ef4852515959" id=".4NXymECcQqz" />
```

Comment:
In the browser, the image shows up correctly, but I have the impression to peek in to a large sheet of paper thru a small hole, because there's a large blank margin around, and I can accidentally un-center it.
Is there a way to crop the image so that it shows up entirely and no shifting is possible?

Reply:
There does not appear to be a way to lock the location of embedded diagrams. I could export a PNG. Perhaps that would be fine in this case because it's a simple diagram.

Reply:
I trimmed the margins of the diagram in https://github.com/trilitech/tezos-developer-docs/pull/287

Reply:
OK, we continue the discussion there.